scripts/helm.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# All HELM related common logic should be kept here
+
+function _set_variables() {
+
+  ## internal func: Sets variables needed for running helm pacakge
+
+  echo "Intialising variables needed for HELM Publish"
+
+  APP_NAME=$(echo $TRAVIS_REPO_SLUG | cut -d"/" -f2)
+
+  if [[ -z "$TRAVIS_TAG" ]]; then
+      VERSION=$(date +%y.%m.%d)-${TRAVIS_COMMIT}
+      APP_VERSION=${TRAVIS_COMMIT}
+  else
+    VERSION=${TRAVIS_TAG}
+    APP_VERSION=${TRAVIS_TAG}
+  fi
+
+}
+
+function _validate_variables() {
+
+  ## internal func: Validates all required variables exist
+
+  echo "Validating input variables needed for HELM Publish"
+
+  # Thorws error if TRAVIS_REPO_SLUG not provided
+  # Expected format: "quid/APP_NAME" eg: "quid/quid_datadog"
+  if [ -z "$TRAVIS_REPO_SLUG" ]; then
+    echo "Error: Must provide TRAVIS_REPO_SLUG"
+    exit 1
+  fi
+
+  # Thorws error if TRAVIS_TAG as well as TRAVIS_COMMIT is not present
+  if [[ -z "$TRAVIS_TAG" ]]; then
+    if [[ -z "$TRAVIS_COMMIT" ]]; then
+      echo "Error: One of TRAVIS_COMMIT OR TRAVIS_TAG must be present"
+      exit 1
+    fi
+  fi
```

Comment:
can we make this generic than for TRAVIS

Reply:
TRAVIS variables are being assigned to local variables before being consumed. 
This way whenever we replace TRAVIS with something we just have to update just set variable code respective alternative source which probably might be JENKINS vars.

```APP_NAME=$(echo $TRAVIS_REPO_SLUG | cut -d"/" -f2)
  if [[ -z "$TRAVIS_TAG" ]]; then
      VERSION=$(date +%y.%m.%d)-${TRAVIS_COMMIT}
      APP_VERSION=${TRAVIS_COMMIT}
  else
    VERSION=${TRAVIS_TAG}
    APP_VERSION=${TRAVIS_TAG}
  fi
``